server/src/node/main.ts

```diff
@@ -26,8 +26,53 @@ export namespace Files {
 	export const resolveModulePath = fm.resolveModulePath;
 }
 
+let _isDetached: boolean | undefined = undefined;
+
+/**
+ * @returns {boolean} True if the process is detached from its parent, false otherwise.
+ *
+ * @description
+ * When a process is detached, it will continue running even if it's parent process
+ * terminates. **But EXIT notifications from the client will be honored regardless.**
+ *
+ * @remarks
+ * This function assumes the parent process has been properly configured with
+ * necessary settings (e.g., using `child_process.unref()`).
+ */
+function isDetached() {
+	// cached result
+	if (_isDetached !== undefined) {
+		return _isDetached;
+	}
+
+	if(process.argv.includes('--detached')) {
+		_isDetached = true;
+
+		// Ignores the default behavior when then parent/child communication
+		// channel (e.g. IPC) disconnects, the child terminates.
+		process.on('disconnect', () => {});
+
+		// Keeps the server alive since node may terminate this process if
```

Comment:
I would not hardcode this here. Could we pass that as a argument to the --detach param.

Reply:
Good point, I've added an optional value for `--detached NNNN` to set the timeout in milliseconds.